vue-gtag.d.ts

```diff
@@ -0,0 +1,73 @@
+declare module 'vue-gtag' {
+  import _Vue, { PluginFunction } from 'vue';
+
+  export interface PageView {
+    page_path: string;
+  }
+
+  export interface Event {
+    event_category: string;
+    event_label: string;
+    value: string;
+  }
+
+  export interface ScreenView {
+    app_name: string;
+    screen_name: string;
+  }
+
+  export interface Purchase {
+    transaction_id: string;
+    affiliation: string;
+    value: number;
+  }
+
+  export interface Linker {
+    domains: string[];
+  }
+
+  export interface Exception {
+    description: string;
+    fatal: boolean;
+  }
+
+  export type Dictionary<T> = { [key: string]: T };
+
+  export interface VueGtag {
+    pageview(pageView: PageView): void;
+    event(action: string, event: Event): void;
+    screenview(screenView: ScreenView): void;
+    customMap(map: Dictionary<string>): void;
+    purchase(purchase: Purchase): void;
+    linker(config: Linker): void;
+    exception(ex: Exception): void;
+  }
+
+  export interface PluginOptions {
+    pageTrackerTemplate?: () => void;
+    onBeforeTrack?: () => void;
+    onAfterTrack?: () => void;
+    onReady?: () => void;
+    enabled?: boolean;
+    disableScriptLoad?: boolean;
+    bootstrap?: boolean;
+    globalObjectName?: string;
+    pageTrackerEnabled?: boolean;
+    pageTrackerScreenviewEnabled: boolean;
+    defaultGroupName?: string;
+    includes: any;
+    config: any;
```

Comment:
I wonder if we can be more precise with this one, something like:

```ts
interface Config {
  id: string,
  params?: {
    [key: string]: any
  }
}
```

not sure if that is exhaustive, i'd have to look myself ^

Reply:
Yep, I'll update the config to this